### ISSUE TITLE
RadarChart PolarAngleAxis getting cut off.

### ISSUE THREAD

**Issue by kenperregaux** [recharts/recharts#4266]
- [ ] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/simple-radar-chart-forked-qfjnvf?file=%2Fsrc%2FApp.tsx&from-embed=)

### Steps to reproduce
Wrapping only happens if there is a space in the label. I would expect it to wrap regardless. Using scaleToFit removes the label entirely with or without a width.

### What is expected?
Wrapping regardless of label containing spaces or scaleToFit to work.

### What is actually happening?
See example above...text wraps only if label contains a space OR scaleToFit to work.

| Environment | Info |
|---|---|
| Recharts | v2.7.3 |
| React | 18.2 |
| System | Windows 11 Pro |
| Browser | Chrome Version  122.0.6261.71 |

**Comment by ckifer** [comment]
I would generally say this is an issue for `margin` on `RadarChart` to solve

[![Edit simple-radar-chart (forked)](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/simple-radar-chart-forked-f786ck?file=%2Fsrc%2FApp.tsx%3A66%2C6)

`margin` however, does not take effect when using static `cy`, `cx`, and `outerRadius`. If you'd like to change those values while still having them be relative, you can change them as percentage values rather than whole numbers.

**Comment by kenperregaux** [comment]
That is fine until you make the chart smaller (say width 350) then you have to play with the margins until it fits? This makes no sense. If I use scaleToFIt or set a width on the Text it should wrap automatically (not just when the label has spaces).

**Comment by ckifer** [comment]
is using percentages an issue?

Recharts doesn't have any magic/dynamic `scaleToFit` solutions for any charts at the moment. `scaleToFit` is supposed to be based off of the `Text` component's `width` property. If you have a chart that changes size margin won't adjust automatically unless you make it dynamic yourself (which works well).

That being said, `Text` itself needs some work it seems (apologies, legacy inheritor of recharts. this functionality has been the same a long time before me). 

For now I would just use something custom like the following or use a `foreignObject` to insert HTML and use CSS for line breaking:
```jsx
        tick={(props) => {
          const { payload, ...rest } = props;
          return <text {...rest}>{payload?.value}</text>;
        }}
```

Here is a sandbox containing some of the not great behavior of the text component
[![Edit text-component-variant-issues](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/simple-radar-chart-forked-tydp3k)

The transform done here is incorrect it seems https://github.com/recharts/recharts/blob/3.x/src/component/Text.tsx#L243

**Comment by kenperregaux** [comment]
@ckifer thanks for the responses. I tried different iterations of using Text with width and scaleToFit and no labels show up when I do that (tweak my original codesandbox above and add width). What I ended up doing is trimming the labels and concating '...' if they were too long.

**Comment by ckifer** [comment]
Thanks for understanding!

If you'd like to contribute feel free! there's a ton of improvements to be had 😅

**Comment by razzeee** [comment]
> The transform done here is incorrect it seems [`3.x`/src/component/Text.tsx#L243](https://github.com/recharts/recharts/blob/3.x/src/component/Text.tsx?rgh-link-date=2024-03-05T19%3A12%3A17Z#L243)

Reading up on this, and I think this should at least be addressed?

It seems like a bracket too many / or a double `/ lineWidth`. But I can't get the demo project to work to verify.

**Comment by ckifer** [comment]
It's been a while since I looked at this I should have written down what was wrong 🙃. It's syntactically correct (no extra bracket). 

Let's do a quick evaluation of it. 

Say width is 20 and line width is 5. 


20/4 = 4

4/5 = 0.8 - so we scale down 0.8

If we removed the last division we would be scaling up 4x which I don't think we'd want so I don't think the answer is removing the division either

**Comment by razzeee** [comment]
Yes, that's exactly what's happening. I should have mentioned before, that I did step though it with a debugger attached.

So that matches my findings.

**Comment by ckifer** [comment]
Would you like to make a PR @razzeee? The change might not perfect "scaleToFit" but definitely better than currently. This must've been some long-time fat finger or auto-formatting issue

**Comment by razzeee** [comment]
I can raise a PR, but I have no idea how to validate it - as I can't get the demo project to actually show any charts :)

**Comment by ckifer** [comment]
Can you use the storybook? The demo project isn't used anymore

**Comment by razzeee** [comment]
I can use the storybook (would be good to document that the demo isn't used anymore - I could have missed that)

But I'm struggling to reproduce the issue with the text story - not sure if different externals are needed to break it. The debugger still shows it computing very low scale values, but it's just unchanged.

**Comment by ckifer** [comment]
> would be good to document that the demo isn't used anymore

Yeah.. there's reasons that it still exists but will add something somewhere about it.

> But I'm struggling to reproduce the issue with the text story 

Hmm, I can reproduce with something like
```
      <Surface width={500} height={500}>
        <Text x={50} y={250} width={50} scaleToFit breakAll>
          {/* this should line break and not need to set width to a crazy number */}
          Some Text that overflows the container and a width and scaleToFit
        </Text>
      </Surface>
```

The story might not be taking props as it should be...

Tbh you could probably easier validate in a unit test where you render something like the above and check the scale value before and after the fix. The scale tranform will be in css attributes

**Comment by ckifer** [comment]
that story needs wrapped in a Surface instead of using ResponsiveContainer. It isn't wrapped in an SVG at the moment which is problematic to say the least

**Comment by razzeee** [comment]
Next problem seems to be, that `width` and `lineWidth` end up being almost the same value, as the `lineWidth` is the ellapsed width, so there is not much to scale.

Still better that it shows at all, will send an PR for that.

**Comment by ckifer** [comment]
Cool, scaleToFit is fixed in the 3.x branch
